### PR TITLE
Use DATA_LOST error for circular JSON Schema Refs

### PR DESCRIPTION
**Pull request by pksunkara**
Also add a fragment to uncaught_error

### CODE REVIEW

**Comment by smizell** [comment]
Looks good. Waiting for tests to pass.

**Comment by smizell** [comment]
@pksunkara getting failing tests that look like they are related to the linter?

**Comment by pksunkara** [comment]
@smizell Updated.

**Comment by smizell** [comment]
@pksunkara I see coverage decreased 3.1%. That seems like a lot. Is the reason because the catch is not tested? How hard would it be to test it?

**Comment by pksunkara** [comment]
@smizell Coverage decreased because I removed a few tests since they work with `auth` only,

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "fury-adapter-swagger",
-  "version": "0.8.0-pre.0",
+  "version": "0.8.0-pre.2",
```
Comment (kylef) [inline package.json]: Since you've changed the version number, should CHANGELOG be updated to reflect this too?

Comment (pksunkara) [inline package.json]: No, these are special pre-releases.

**Pull request merged**